### PR TITLE
fix: disable telemetry for executors

### PR DESCRIPTION
**Pull request by jupyterjazz**
Disable telemetry for executors during CI runs
Closes https://github.com/jina-ai/now/issues/720

---

- [x] This PR references an open issue
- [x] Test added
- [ ] Documentation adjusted

### CODE REVIEW

**Comment by joschkabraun** [comment]
Can you add a test which checks that the environment variables for telemetry are correctly set after `app.setup()`.

File: now/app/base/app.py [hunk 1]
```diff
@@ -236,6 +234,8 @@ def setup(
                             },
                         },
                     )
+
```
Comment (azayz) [inline now/app/base/app.py:237]: why did this change place ?

Comment (jupyterjazz) [reply now/app/base/app.py:237]: because the autocomplete executor was added afterwards

File: tests/unit/apps/test_apps.py [hunk 2]
```diff
@@ -26,3 +30,28 @@ def test_split_text_preprocessing():
     assert len(new_da) == 1
     assert len(new_da[0].chunks) == 1
     assert len(new_da[0].chunks[0].chunks) == 2
+
+
+@pytest.mark.parametrize('disable', [False, True])
+def test_disable_telemetry(disable):
```
Comment (florian-hoenicke) [inline tests/unit/apps/test_apps.py:36]: Is this test passing without your changes on the base app?

Comment (jupyterjazz) [reply tests/unit/apps/test_apps.py:36]: no it wouldn't

File: now/constants.py [hunk 3]
```diff
@@ -8,11 +8,10 @@
 DEMO_DATASET_DOCARRAY_VERSION = '0.13.17'
 # ----------------------------------
 DOCKER_BFF_PLAYGROUND_TAG = '0.0.135-test-reactivate'
-# ----------------------------------
-NOW_PREPROCESSOR_VERSION = '0.0.111-refactor-cleanup-data-loading-3'
-NOW_QDRANT_INDEXER_VERSION = '0.0.10-refactor-remove-redundancy-on-ci-18'
-NOW_ELASTIC_INDEXER_VERSION = '0.0.9-refactor-remove-redundancy-on-ci-18'
-NOW_AUTOCOMPLETE_VERSION = '0.0.5-refactor-remove-redundancy-on-ci-18'
+NOW_PREPROCESSOR_VERSION = '0.0.109-fix-telemetry-4'
+NOW_QDRANT_INDEXER_VERSION = '0.0.9-refactor-improve-performance-curation-3'
```
[the remark this answers is not in the document]
Comment (jupyterjazz) [reply now/constants.py:12]: I haven't added `0.0.9-refactor-improve-performance-curation-3` myself, it was merged from main.
Preprocessor updated.

File: tests/unit/apps/test_apps.py [hunk 4]
```diff
@@ -26,3 +30,28 @@ def test_split_text_preprocessing():
...
+    if disable:
+        os.environ['JINA_OPTOUT_TELEMETRY'] = 'disableTelemetry'
+    else:
+        os.environ.pop('JINA_OPTOUT_TELEMETRY', None)
+
```
Comment (joschkabraun) [inline tests/unit/apps/test_apps.py:41]: as discussed in pairing, this has has side effects to other tests run in the same runner. We can:
- set the environment variable value in the end again

**Review by joschkabraun — APPROVED** [review]
LGTM! Only small comment

**Review by florian-hoenicke — APPROVED** [review]
LGTM

**Pull request merged**